pymongo_auth_aws/auth.py

```diff
@@ -45,6 +47,26 @@ def _aws_temp_credentials():
     if access_key and secret_key:
         return AwsCredential(
             access_key, secret_key, os.environ.get('AWS_SESSION_TOKEN'))
+    # Check if environment variables exposed by IRSA are present
+    irsa_web_id_file = os.getenv('AWS_WEB_IDENTITY_TOKEN_FILE')
+    irsa_role_arn = os.getenv('AWS_ROLE_ARN')
+    if irsa_web_id_file and irsa_role_arn:
+        print("in irsa flow with values: ", irsa_web_id_file, irsa_role_arn)
+        try:
+            with open(irsa_web_id_file) as f:
+                irsa_web_id_token = f.read()
+            access_key, secret_key, session_token = _irsa_assume_role(
+                irsa_role_arn,
+                irsa_web_id_token,
+                'pymongo-auth-aws'
+            )
+        except ClientError as error:
+            error_message = error.response['Error']['Message']
```

Comment:
Are all ClientErrors guaranteed to contain `.response['Error']['Message']`?

Reply:
I believe it is more robust to use BotoCoreError or even better just plain Exception here. boto3 and botocore seem like they can raise many different non-ClientError types here.

Reply:
Perhaps two `except` blocks? One for `ClientError` to make a best-effort attempt at a more helpful exception, and one for generic `Exceptions` when best-effort fails?

Reply:
The `error_message = error.response['Error']['Message']` logic isn't needed so we can use a single `except Exception`. Simply converting the ClientError to a string provides a better error message:
```
>>> error.response['Error']['Message']
"User: arn:aws:iam::557821124784:user/mongo-python-driver-key is not authorized to perform: kinesis:DescribeStream on resource: arn:aws:kinesis:us-east-1:557821124784:stream/myDataStream because no identity-based policy allows the kinesis:DescribeStream action"

>>> str(error)
"An error occurred (AccessDeniedException) when calling the DescribeStream operation: User: arn:aws:iam::557821124784:user/mongo-python-driver-key is not authorized to perform: kinesis:DescribeStream on resource: arn:aws:kinesis:us-east-1:557821124784:stream/myDataStream because no identity-based policy allows the kinesis:DescribeStream action"
```

Reply:
Good to know, will change.

Reply:
Changed, please have another look.